packages/angular_devkit/core/src/analytics/forwarder.ts

```diff
@@ -0,0 +1,131 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { JsonObject } from '../json';
+import { Analytics, EventOptions, PageviewOptions, ScreenviewOptions, TimingOptions } from './api';
+
+
+export enum AnalyticsReportKind {
+  Event = 'event',
+  Screenview = 'screenview',
+  Pageview = 'pageview',
+  Timing = 'timing',
+}
+
+export interface AnalyticsReportBase extends JsonObject {
+  kind: AnalyticsReportKind;
+}
+
+export interface AnalyticsReportEvent extends AnalyticsReportBase {
+  kind: AnalyticsReportKind.Event;
+  options: JsonObject & EventOptions;
+  c: string;
```

Comment:
Is there a good reason to leave these properties as single letter instead of a real name? If so, it should be mentioned somewhere.

Reply:
No really. They're are stringified and parsed to/from JSON, but I don't think it makes enough of a difference. Fixed.